src/ucp/core/ucp_am.c

```diff
@@ -912,23 +892,28 @@ ucp_am_try_send_short(ucp_ep_h ep, uint16_t id, uint32_t flags,
                       const void *header, size_t header_length,
                       const void *buffer, size_t length)
 {
-    if (ucs_unlikely(((length != 0) && (header_length != 0)) ||
-                     (flags & UCP_AM_SEND_FLAG_RNDV))) {
+    size_t total_length;
+
+    if (ucs_unlikely(flags & UCP_AM_SEND_FLAG_RNDV)) {
         goto out;
     }
 
-    if (ucs_unlikely(!ucp_proto_is_inline(ep,
-                                          &ucp_ep_config(ep)->am_u.max_eager_short,
-                                          length + header_length))) {
+    total_length = header_length + length;
+
+    if (ucs_unlikely(!ucp_proto_is_inline(
```

Comment:
this is not needed anymore, because you are checking `ucp_proto_is_inline` for REPLY and non-REPLY below

Reply:
`ucp_proto_is_inline` is called with different arguments below. Thus the statement could not be removed without changing the logic. But the statement at line 913 can be removed. Because it always true after passing the check at the line 903.

Reply:
> `ucp_proto_is_inline` is called with different arguments below.

yes, right
I meant that we don't need this, but the statements below should be ok:
- one for REPLY AM
- another one for non-REPLY AM

Reply:
Current behavior:

```
if (!ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_eager_short...)) {
    goto out;
}
if ((flags & UCP_AM_SEND_REPLY) && ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_reply_eager_short...)) {
    return ucp_am_send_short(...1);
}
return ucp_am_send_short(...0);
```

is not equal to:

```
if ((flags & UCP_AM_SEND_REPLY) && ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_reply_eager_short...)) {
    return ucp_am_send_short(...1);
} else if (ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_eager_short...)) {
    return ucp_am_send_short(...0);
}
```

if `ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_eager_short...)` returns false, but
`(flags & UCP_AM_SEND_REPLY) && ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_reply_eager_short...)` is true.

Reply:
yes, I understand
then it means that if REPLY can be used, we will fail the:
```
if (!ucp_proto_is_inline(...&ucp_ep_config(ep)->am_u.max_eager_short...)) { goto out; }
```
which is not expected

Reply:
if it is expected, then need to remove `max_reply_eager_short`

Reply:
@brminich removed the double checking the same limit + fixed sending REPLY